### PR TITLE
Adding json path functions to extract values from json object

### PR DESCRIPTION
**Pull request by fx19880617**
## Description
Adding json extract scalar functions to help extract messages from json object.

This can help users extract values from json object and flat events during ingestion time.

Sample ingestion config inside the table config: 
```
...
  "ingestionConfig": {
    "transformConfigs": [
    {
      "columnName": "created_ts",
      "transformFunction": "fromDateTime(created_at, \"yyyy-MM-dd'T'HH:mm:ssZ\")"
    },
    {
      "columnName": "actor_id",
      "transformFunction": "jsonPathLong(actor, '$.id')"
    },
    {
      "columnName": "actor_login",
      "transformFunction": "jsonPathString(actor, '$.login')"
    },
    {
      "columnName": "actor_display_login",
      "transformFunction": "jsonPathString(actor, '$.display_login')"
    },
    {
      "columnName": "actor_gravatar_id",
      "transformFunction": "jsonPathString(actor, '$.gravatar_id')"
    },
    {
      "columnName": "actor_url",
      "transformFunction": "jsonPathString(actor, '$.url')"
    },
    {
      "columnName": "actor_avatar_url",
      "transformFunction": "jsonPathString(actor, '$.avatar_url')"
    },
    {
      "columnName": "repo_id",
      "transformFunction": "jsonPathLong(repo, '$.id')"
    },
    {
      "columnName": "repo_name",
      "transformFunction": "jsonPathString(repo, '$.name')"
    },
    {
      "columnName": "repo_url",
      "transformFunction": "jsonPathString(repo, '$.url')"
    },
    {
      "columnName": "payload_str",
      "transformFunction": "jsonFormat(payload)"
    }
    ]
  }
...
```

### CODE REVIEW

**Comment by siddharthteotia** [comment]
Can we add tests/examples to demonstrate the usage of new functions?

**Comment by fx19880617** [comment]
> Can we add tests/examples to demonstrate the usage of new functions?

done

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/JsonFunctions.java [hunk 1]
```diff
@@ -56,4 +57,42 @@ public static String jsonFormat(Object object)
...
+  public static String jsonPathString(Object object, String jsonPath)
+      throws JsonProcessingException {
+    return JsonUtils.objectToString(jsonPath(object, jsonPath));
+  }
+
+  /**
+   * Extract from Json with path to Long
+   */
+  @ScalarFunction
+  public static Long jsonPathLong(Object object, String jsonPath)
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/function/scalar/JsonFunctions.java:85]: return primitive long

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/JsonFunctions.java [hunk 2]
```diff
@@ -56,4 +57,42 @@ public static String jsonFormat(Object object)
...
+      throws JsonProcessingException {
+    return Long.parseLong(jsonPathString(object, jsonPath));
+  }
+
+  /**
+   * Extract from Json with path to Double
+   */
+  @ScalarFunction
+  public static Double jsonPathDouble(Object object, String jsonPath)
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/function/scalar/JsonFunctions.java:94]: return primitive double

File: pinot-common/src/test/java/org/apache/pinot/common/function/JsonFunctionsTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        "    \"ref_type\": \"branch\"," +
+        "    \"master_branch\": \"master\"," +
+        "    \"description\": null," +
+        "    \"pusher_type\": \"user\"" +
+        "  }," +
+        "  \"public\": true," +
+        "  \"created_at\": \"2018-01-01T11:12:53Z\"" +
+        "}";
+    assertEquals(JsonFunctions.jsonPathString(jsonString, "$.actor.id"), "33500718");
+    assertEquals(JsonFunctions.jsonPathLong(jsonString, "$.actor.id"), Long.valueOf(33500718L));
```
Comment (Jackie-Jiang) [inline pinot-common/src/test/java/org/apache/pinot/common/function/JsonFunctionsTest.java:61]:
```suggestion
    assertEquals(JsonFunctions.jsonPathLong(jsonString, "$.actor.id"), 33500718L);
```

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**